### PR TITLE
#1951 and #1952

### LINKED ISSUE #1951: [BUG] Loading COCO-formatted data does not work
### System information

-   **OS Platform and Distribution: Linux Ubuntu 18.04**:
-   **FiftyOne installed from pip**:
-   **FiftyOne version: FiftyOne v0.16.5, Voxel51, Inc.**:
-   **Python version: Python 3.8.10**:

### Commands to reproduce

Pip install the fiftyone and just follow the tutorial: https://voxel51.com/docs/fiftyone/integrations/coco.html#loading-coco-formatted-data

```py
import os

import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart")

# Give the dataset a classes list so it can be exported + imported
dataset.default_classes = dataset.distinct("ground_truth.detections.label")

# The directory in which the dataset's images are stored
IMAGES_DIR = os.path.dirname(dataset.first().filepath)

# Export some labels in COCO format
dataset.take(5).export(
    dataset_type=fo.types.COCODetectionDataset,
    label_field="ground_truth",
    labels_path="/tmp/coco.json",
)

# Load COCO formatted dataset
coco_dataset = fo.Dataset.from_dir(
    dataset_type=fo.types.COCODetectionDataset,
    data_path=IMAGES_DIR,
    labels_path="/tmp/coco.json",
    include_id=True,
    label_field="",
)
```

### Describe the problem

Can't load COCO formatted data. Following the code in the documentation I get this error:

```
ValueError: Invalid field name: '_detections'. Field names cannot start with '_'
```

### Code to reproduce issue

Clean install the fiftyone and just follow the tutorial: https://voxel51.com/docs/fiftyone/integrations/coco.html#loading-coco-formatted-data

### Other info / logs
Full error log:

```
Loading 'quickstart'
 100% |███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 200/200 [3.1s elapsed, 0s remaining, 59.3 samples/s]      
Dataset 'quickstart' created
 100% |███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [78.0ms elapsed, 0s remaining, 64.1 samples/s] 
   0% ||--------------------------------------------------------------------------------------------------------------------------------------| 0/5 [498.6us elapsed, ? remaining, ? samples/s] 
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [1], in <cell line: 22>()
     15 dataset.take(5).export(
     16     dataset_type=fo.types.COCODetectionDataset,
     17     label_field="ground_truth",
     18     labels_path="/tmp/coco.json",
     19 )
     21 # Load COCO formatted dataset
---> 22 coco_dataset = fo.Dataset.from_dir(
     23     dataset_type=fo.types.COCODetectionDataset,
     24     data_path=IMAGES_DIR,
     25     labels_path="/tmp/coco.json",
     26     include_id=True,
     27     label_field="",
     28 )

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/dataset.py:3811, in Dataset.from_dir(cls, dataset_dir, dataset_type, data_path, labels_path, name, label_field, tags, **kwargs)
   3722 """Creates a :class:`Dataset` from the contents of the given directory.
   3723 
   3724 You can create datasets with this method via the following basic
   (...)
   3808     a :class:`Dataset`
   3809 """
   3810 dataset = cls(name)
-> 3811 dataset.add_dir(
   3812     dataset_dir=dataset_dir,
   3813     dataset_type=dataset_type,
   3814     data_path=data_path,
   3815     labels_path=labels_path,
   3816     label_field=label_field,
   3817     tags=tags,
   3818     **kwargs,
   3819 )
   3820 return dataset

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/dataset.py:2591, in Dataset.add_dir(self, dataset_dir, dataset_type, data_path, labels_path, label_field, tags, expand_schema, add_info, **kwargs)
   2490 """Adds the contents of the given directory to the dataset.
   2491 
   2492 You can perform imports with this method via the following basic
   (...)
   2580     a list of IDs of the samples that were added to the dataset
   2581 """
   2582 dataset_importer, _ = foud.build_dataset_importer(
   2583     dataset_type,
   2584     dataset_dir=dataset_dir,
   (...)
   2588     **kwargs,
   2589 )
-> 2591 return self.add_importer(
   2592     dataset_importer,
   2593     label_field=label_field,
   2594     tags=tags,
   2595     expand_schema=expand_schema,
   2596     add_info=add_info,
   2597 )

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/dataset.py:3138, in Dataset.add_importer(self, dataset_importer, label_field, tags, expand_schema, add_info)
   3096 def add_importer(
   3097     self,
   3098     dataset_importer,
   (...)
   3102     add_info=True,
   3103 ):
   3104     """Adds the samples from the given
   3105     :class:`fiftyone.utils.data.importers.DatasetImporter` to the dataset.
   3106 
   (...)
   3136         a list of IDs of the samples that were added to the dataset
   3137     """
-> 3138     return foud.import_samples(
   3139         self,
   3140         dataset_importer,
   3141         label_field=label_field,
   3142         tags=tags,
   3143         expand_schema=expand_schema,
   3144         add_info=add_info,
   3145     )

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/utils/data/importers.py:129, in import_samples(dataset, dataset_importer, label_field, tags, expand_schema, add_info)
    126     num_samples = None
    128 samples = map(parse_sample, iter(dataset_importer))
--> 129 sample_ids = dataset.add_samples(
    130     samples, expand_schema=expand_schema, num_samples=num_samples
    131 )
    133 if add_info and dataset_importer.has_dataset_info:
    134     info = dataset_importer.get_dataset_info()

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/dataset.py:1514, in Dataset.add_samples(self, samples, expand_schema, validate, num_samples)
   1512 sample_ids = []
   1513 with fou.ProgressBar(total=num_samples) as pb:
-> 1514     for batch in batcher:
   1515         sample_ids.extend(
   1516             self._add_samples_batch(batch, expand_schema, validate)
   1517         )
   1518         pb.update(count=len(batch))

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/utils.py:929, in DynamicBatcher.__next__(self)
    927     idx = 0
    928     while idx < batch_size:
--> 929         batch.append(next(self._iter))
    930         idx += 1
    932 except StopIteration:

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/utils/data/importers.py:411, in _build_parse_sample_fcn.<locals>.parse_sample(sample)
    404 sample = Sample(
    405     filepath=image_path,
    406     metadata=image_metadata,
    407     tags=tags,
    408 )
    410 if isinstance(label, dict):
--> 411     sample.update_fields(
    412         {label_key(k): v for k, v in label.items()}
    413     )
    414 elif label is not None:
    415     sample[label_field] = label

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/document.py:202, in _Document.update_fields(self, fields_dict, expand_schema)
    189 """Sets the dictionary of fields on the document.
    190 
    191 Args:
   (...)
    199         exist
    200 """
    201 for field_name, value in fields_dict.items():
--> 202     self.set_field(field_name, value, create=expand_schema)

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/sample.py:96, in _SampleMixin.set_field(self, field_name, value, create)
     93     self.frames.update(value)
     94     return
---> 96 super().set_field(field_name, value, create=create)

File ~/Downloads/venv/lib/python3.8/site-packages/fiftyone/core/document.py:181, in _Document.set_field(self, field_name, value, create)
    169 """Sets the value of a field of the document.
    170 
    171 Args:
   (...)
    178     AttirubteError: if the field does not exist and ``create == False``
    179 """
    180 if field_name.startswith("_"):
--> 181     raise ValueError(
    182         "Invalid field name: '%s'. Field names cannot start with '_'"
    183         % field_name
    184     )
    186 self._doc.set_field(field_name, value, create=create)

ValueError: Invalid field name: '_detections'. Field names cannot start with '_'
```

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core `fiftyone` Python library issue
-   [ ] `Server`: Fiftyone server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently.
-   [x] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community.
-   [ ] No. I cannot contribute a bug fix at this time.

### LINKED ISSUE #1952: [BUG] Labels are swapped after exporting splits (test, train, val) in YOLOv5 format
### System information

-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
-   **FiftyOne installed from (pip or source)**: pip
-   **FiftyOne version (run `fiftyone --version`)**: FiftyOne v0.16.5, Voxel51, Inc.
-   **Python version**: Python 3.8.10

### Commands to reproduce
1. Import libraries
```python
import fiftyone.utils.splits as fous
import fiftyone as fo
from pathlib import Path
```
2. Load a COCO formatted dataset (not split)
```python
coco_path = Path("datasets/coco_custom_v1.1")
coco_dataset = fo.Dataset.from_dir(
        dataset_type=fo.types.COCODetectionDataset,
        data_path=coco_path/"data",
        labels_path=coco_path/"labels.json"
)
```
3. Start the fiftyone server and make sure the dataset labels look good
```python 
session = fo.launch_app(coco_dataset, remote=True, port=6262)
```
4. Create split tags
```python
fous.random_split(coco_dataset, {"train": 0.65, "test": 0.2, "val": 0.15})
```
5. ***BUG HERE*** -> Export all the splits to YoloV5 format
```python
for split in ["train", "test", "val"]:
    split_view = coco_dataset.match_tags(split)
    split_view.export(
        export_dir="datasets/test-split-yolo",
        dataset_type=fo.types.YOLOv5Dataset,
        split=split
    )
```

6. Import the dataset we just exported to look at the labels
```python
coco_split_imported_dataset = fo.Dataset("coco_split_imported_dataset")
for split in ["train", "test", "val"]:
coco_split_imported_dataset.add_dir(
            dataset_dir="datasets/test-split-yolo",
            dataset_type=fo.types.YOLOv5Dataset,
            split=split,
            tags=split,
)
```
7. ***Observe that for the samples with label `'train'` the class labels are swapped***

### Describe the problem

I have a task for object detection with 2 classes. After training a Yolo model I noticed that my confusion matrix looks weird, it was performing poorly on the same classes but well on the opposite classes, meaning in my validation set labels were swapped. After hours of debugging I pinned it down. Fiftyone has a bug. When you split a dataset and then export it into YOLOv5 format, it messes up the labels (swaps them) for train set.

### Code to reproduce issue

Described in the `Commands to reproduce` section in detail. 

### Other info / logs

My dataset has 2 classes.

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core `fiftyone` Python library issue
-   [ ] `Server`: Fiftyone server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another
member of your organization be willing to contribute a fix for this bug to the
FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently.
-   [x] Yes. I would be willing to contribute a fix for this bug with guidance
        from the FiftyOne community.
-   [ ] No. I cannot contribute a bug fix at this time.

### PR DESCRIPTION
**Pull request by brimoor**
Resolves #1951
Resolves #1952 

#1951 is resolved by simply removing a stray `label_field=""` from the docs.

#1952 is resolved by adding an informative error message when trying to export multiple YOLOv5 splits without providing the `classes` parameter to ensure a common class list is used.

```py
import fiftyone as fo
import fiftyone.zoo as foz
import fiftyone.utils.splits as fous

dataset = foz.load_zoo_dataset("quickstart")

dataset = dataset.select_fields("ground_truth").clone()
dataset.untag_samples("validation")

classes = dataset.distinct("ground_truth.detections.label")

fous.random_split(dataset, {"train": 0.65, "test": 0.2, "val": 0.15})

for split in ["train", "test", "val"]:
    dataset.match_tags(split).export(
        export_dir="/tmp/yolov5",
        dataset_type=fo.types.YOLOv5Dataset,
        classes=classes,  # you'll see an error if you omit this
        split=split,
    )

dataset2 = fo.Dataset()
for split in ["train", "test", "val"]:
    dataset2.add_dir(
        dataset_dir="/tmp/yolov5",
        dataset_type=fo.types.YOLOv5Dataset,
        split=split,
        tags=split,
    )

session = fo.launch_app(dataset2)
```

### CODE REVIEW

**Comment by brimoor** [comment]
@ehofesmann in this PR, I also removed a couple stray references to how `classes`/`default_classes` are used from the docs, which are no longer true as of https://github.com/voxel51/fiftyone/pull/1858.

There's one more reference to `default_classes` in the [Lightning Flash docs](https://github.com/voxel51/fiftyone/blob/develop/docs/source/integrations/lightning_flash.rst). Is that still accurate? Or should `classes` be manually provided as an argument to `from_fiftyone()` in light of https://github.com/voxel51/fiftyone/pull/1858?

Note that one of the motivations for https://github.com/voxel51/fiftyone/pull/1858 was this issue: https://github.com/Lightning-AI/lightning-flash/issues/763. Basically I think it's a better behavior for `classes` to always be manually provided.

I guess if any code needs to be changed, that would be happening over in Lightning Flash, and our docs could only be updated after that is released...

tldr is there anything relevant to add to this PR?

**Comment by ehofesmann** [comment]
> 



> @ehofesmann in this PR, I also removed a couple stray references to how `classes`/`default_classes` are used from the docs, which are no longer true as of #1858.
> 
> There's one more reference to `default_classes` in the [Lightning Flash docs](https://github.com/voxel51/fiftyone/blob/develop/docs/source/integrations/lightning_flash.rst). Is that still accurate? Or should `classes` be manually provided as an argument to `from_fiftyone()` in light of #1858?
> 
> Note that one of the motivations for #1858 was this issue: [Lightning-AI/lightning-flash#763](https://github.com/Lightning-AI/lightning-flash/issues/763). Basically I think it's a better behavior for `classes` to always be manually provided, to avoid "surprises".
> 
> I guess if any code needs to be changed, that would be happening over in Lightning Flash, and our docs could only be updated after that is released...?
> 
> tldr is there anything relevant to add to this PR?

You're right, to remove that, Lightning Flash will need to be updated. See https://github.com/voxel51/fiftyone/issues/1992 for more details regarding that. Nothing else to be done in this PR.

**Review by ehofesmann — APPROVED** [review]
LGTM, noting the lightning flash reference to `default_classes` will need to be handled according to #1992

**Comment by brimoor** [comment]
@ehofesmann awesome, thanks for the confirmation 💪
